light_curves/ML_AGNzoo.md

```diff
@@ -4,115 +4,159 @@ jupytext:
 import matplotlib.pyplot as plt
 import matplotlib as mpl
 import matplotlib.cm as cm
+from matplotlib.colors import LinearSegmentedColormap
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
 import numpy as np
 import pandas as pd
 sys.path.append('code_src/')
 from data_structures import MultiIndexDFObject
-from ML_utils import unify_lc, stat_bands, autopct_format, combine_bands,\
+from ML_utils import unify_lc, unify_lc_gp, stat_bands, autopct_format, combine_bands,\
 mean_fractional_variation, normalize_mean_objects, normalize_max_objects, \
-normalize_clipmax_objects, shuffle_datalabel, dtw_distance, stretch_small_values_arctan
-from sample_lc import build_sample
+normalize_clipmax_objects, shuffle_datalabel, dtw_distance, stretch_small_values_arctan, translate_bitwise_sum_to_labels, update_bitsums
 from sklearn.cluster import KMeans
 from sklearn.preprocessing import StandardScaler
-try:
-    import umap
-    import umap.plot
-except:
-    !pip install umap-learn[plot]
+from collections import Counter,OrderedDict
+
+import umap
+from sompy import * #using the SOMPY package from https://github.com/sevamoo/SOMPY
+
+import logging
+
+# Get the root logger
+logger = logging.getLogger()
+logger.setLevel(logging.ERROR)
 
 import warnings
 warnings.filterwarnings('ignore')
 
 plt.style.use('bmh')
 colors = [
-    "#6C907D",  # Soft Green
-    "#B9C3C2",  # Pale Gray
-    "#8C9EFF",  # Periwinkle Blue
-    "#FF6347",  # Tomato Red
-    "#66CDAA",  # Medium Aquamarine
-    "#E6E6FA",  # Lavender
-    "#FAFAD2",  # Light Goldenrod Yellow
-    "#FFD700",  # Gold
+    "#3F51B5",  # Ultramarine Blue
+    "#003153",  # Prussian Blue
+    "#0047AB",  # Cobalt Blue
+    "#40826D",  # Viridian Green
+    "#50C878",  # Emerald Green
+    "#FFEA00",  # Chrome Yellow
+    "#CC7722",  # Yellow Ochre
+    "#E34234",  # Vermilion
+    "#E30022",  # Cadmium Red
+    "#D68A59",  # Raw Sienna
+    "#8A360F",  # Burnt Sienna
+    "#826644",  # Raw Umber
 ]
 
-# To build a sample and generate the multiindex lc dataframe the code below is used, which takes long:
-#!python sample_lc.py
-#c,l = build_sample() # if coordinates and labels of the sample are needed separately
+custom_cmap = LinearSegmentedColormap.from_list("custom_theme", colors[1:])
 ```
 
+***
+
+
+## 1) Loading data
 Here we load a parquet file of light curves generated using the multiband_lc notebook. One can build the sample from different sources in the literature and grab the data from archives of interes.
 
 ```{code-cell} ipython3
-parquet_loadname = 'output/df_lc_5k.parquet.gzip'
-parallel_df_lc = MultiIndexDFObject()
-parallel_df_lc.data = pd.read_parquet(parquet_loadname)
+#sample_table = Table.read('data/agnsample_feb7.ecsv', format="ascii.ecsv") # if needed, contains coordinates, redshift and all labels
+df = pd.read_parquet('data/df_lc_020724.parquet.gzip')
+
+df2 = df[df.index.get_level_values('label') != '64'] # remove 64 for SPIDER only as its too large
```

Comment:
Somewhere above this line you should describe what you are doing with the bitwise sums so that "removing 64" has a meaning.